test/unit/spec/common/jobs-Spec.js

```diff
@@ -100,20 +120,73 @@ define([
                 });
             });
         });
-        const mapping = {
-            [jcm.MESSAGE_TYPE.STATUS]: 'isValidBackendJobStateObject',
-            [jcm.MESSAGE_TYPE.INFO]: 'isValidJobInfoObject',
-            [jcm.MESSAGE_TYPE.LOGS]: 'isValidJobLogsObject',
-            [jcm.MESSAGE_TYPE.RETRY]: 'isValidJobRetryObject',
-        };
 
-        Object.keys(ResponseData).forEach((respType) => {
-            const fn = mapping[respType];
-            describe(`the ${fn} function, contd`, () => {
-                Object.values(ResponseData[respType]).forEach((obj) => {
-                    it('asserts the validity of ' + JSON.stringify(obj), () => {
-                        expect(Jobs[fn](obj)).toBeTrue();
+        describe('the validateMessage function', () => {
+            Object.keys(ResponseData).forEach((respType) => {
+                it(`should validate a multi-job ${respType} message`, () => {
+                    expect(
+                        Jobs.validateMessage({
+                            message: ResponseData[respType],
+                            type: respType,
+                        })
+                    ).toEqual({
+                        valid: ResponseData[respType],
+                    });
+                });
+            });
+
+            ['job-status', 'greetings', 'error', 'NEW', jcm.MESSAGE_TYPE.CANCEL].forEach((type) => {
+                it(`should reject invalid message type ${type}`, () => {
+                    expect(Jobs.validateMessage({ message: null, type })).toEqual({
+                        invalid: null,
+                    });
+                });
+            });
+
+            invalidTypes.forEach((dataType) => {
+                it(`should reject message with an invalid data type ${Utils.objectToString(
+                    dataType
+                )}`, () => {
+                    expect(
+                        Jobs.validateMessage({ message: dataType, type: jcm.MESSAGE_TYPE.INFO })
+                    ).toEqual({ invalid: dataType });
+                });
+            });
+
+            const gotValidator = ['INFO', 'LOGS', 'RETRY', 'RUN_STATUS', 'STATUS', 'STATUS_ALL'];
```

Comment:
these are all the message types that validators exist for. This is not the most elegant way of doing things, but since it's just for testing...